HSEARCH-2545 Validation of IndexedEmbedded targeting a numeric document ID fails inside a non-indexed entity

yrodiere opened the pull request:
https://hibernate.atlassian.net/browse/HSEARCH-2545

See the commit comments for details about the solution.

gsmet commented on orm/src/test/java/org/hibernate/search/test/engine/NumericDocumentIdIndexedEmbeddedTest.java:
```diff
@@ -0,0 +1,122 @@
+
+	@After
+	public void cleanData() {
+		if ( getTestResourceManager() == null ) {
```
Totally unrelated but couldn't we factorize this somehow? I mean a method taking the classes as parameters you directly call in your `@After`.

yrodiere replied:
Actually, this code is probably not even needed, since the default HBM2DDL strategy drops the schema when closing the factory. I'll remove the code and check that we really drop the schema.

yrodiere replied:
@gsmet I just checked: the tables are dropped automatically after the test. So I removed this code. 
I should take more care when I copy/paste...

Sanne commented on engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java:
```diff
@@ -1040,7 +1040,9 @@ private void initializeMemberLevelAnnotations(String prefix,
 		checkForContainedIn( member, typeMetadataBuilder, parseContext );
 
-		numericFields.validate();
+		if ( !parseContext.skipFieldBridges() ) {
```
Puzzling check. I guess I could find out the reasons by debugging the tests, but this means some comments here could be helpful?

yrodiere replied:
It's definitely not the only puzzling thing in this part of the code :D But yes, you're right.  I'll add a comment :)

gsmet replied:
@Sanne it's probably something we should discuss for 6. The annotation metadata provider magic became a little too magic and obscure lately with Es and obscure bug fixing.

yrodiere commented:
@Sanne @gsmet Thanks for the reviews. I think I addressed your concerns, is it ok to merge as is?

The additional comment is there: https://github.com/hibernate/hibernate-search/pull/1285/files#diff-0a572dae381a036d0fc969a5b45a2ac7R1043

gsmet commented:
@yrodiere +1

yrodiere commented:
Jenkins, retest this please

yrodiere commented:
Jenkins, retest this please
